src/components/button-toggle/button-toggle.spec.ts

```diff
@@ -0,0 +1,341 @@
+      testComponent.isGroupDisabled = true;
+      fixture.detectChanges();
+
+      buttonToggleNativeElements[0].click();
+      expect(buttonToggleInstances[0].checked).toBe(false);
+    });
+
+    it('should update the group value when one of the toggles changes', () => {
+      expect(groupInstance.value).toBeFalsy();
+      let nativeCheckboxLabel = buttonToggleDebugElements[0].query(By.css('label')).nativeElement;
+
+      nativeCheckboxLabel.click();
+      fixture.detectChanges();
+
+      expect(groupInstance.value).toBe('test1');
+      expect(groupInstance.selected).toBe(buttonToggleInstances[0]);
+    });
+
+    it('should update the group and toggles when one of the button toggles is clicked', () => {
+      expect(groupInstance.value).toBeFalsy();
+      let nativeCheckboxLabel = buttonToggleDebugElements[0].query(By.css('label')).nativeElement;
+
+      nativeCheckboxLabel.click();
+      fixture.detectChanges();
+
+      expect(groupInstance.value).toBe('test1');
+      expect(groupInstance.selected).toBe(buttonToggleInstances[0]);
+      expect(buttonToggleInstances[0].checked).toBe(true);
+      expect(buttonToggleInstances[1].checked).toBe(false);
+
+      let nativeCheckboxLabel2 = buttonToggleDebugElements[1].query(By.css('label')).nativeElement;
+
+      nativeCheckboxLabel2.click();
+      fixture.detectChanges();
+
+      expect(groupInstance.value).toBe('test2');
+      expect(groupInstance.selected).toBe(buttonToggleInstances[1]);
+      expect(buttonToggleInstances[0].checked).toBe(false);
+      expect(buttonToggleInstances[1].checked).toBe(true);
+    });
+
+    it('should check a button toggle upon interaction with underlying native radio button', () => {
+      let nativeRadioInput = buttonToggleDebugElements[0].query(By.css('input')).nativeElement;
+
+      nativeRadioInput.click();
+      fixture.detectChanges();
+
+      expect(buttonToggleInstances[0].checked).toBe(true);
+      expect(groupInstance.value);
+    });
+
+    it('should emit a change even from button toggles', fakeAsync(() => {
+      expect(buttonToggleInstances[0].checked).toBe(false);
+
+      let changeSpy = jasmine.createSpy('button-toggle change listener');
+      buttonToggleInstances[0].change.subscribe(changeSpy);
+
+      buttonToggleInstances[0].checked = true;
+      fixture.detectChanges();
+      tick();
+      expect(changeSpy).toHaveBeenCalled();
+
+      buttonToggleInstances[0].checked = false;
+      fixture.detectChanges();
+      tick();
+      expect(changeSpy).toHaveBeenCalledTimes(2);
+    }));
+
+    it('should emit a change event from the button toggle group', fakeAsync(() => {
+      expect(groupInstance.value).toBeFalsy();
+
+      let changeSpy = jasmine.createSpy('button-toggle-group change listener');
+      groupInstance.change.subscribe(changeSpy);
+
+      groupInstance.value = 'test1';
+      fixture.detectChanges();
+      tick();
+      expect(changeSpy).toHaveBeenCalled();
```

Comment:
Do we want to be more specific  here?  ie. `expect(changeSpy).toHaveBeenCalledWith('test1')`